rfcs/build-system/04-build-shield.md

```diff
@@ -0,0 +1,107 @@
+- Lack of recognition/appreciation for time spent and value provided to the team
+- Build-related knowledge stays concentrated with a very limited number of people
+- Harder to recognize patterns of failures
+
+Having build shield as a desginated, rotating responsibility will help address all of these issues. It will also help more people on the team gain experience with build troubleshooting.
+
+## Detailed Design or Proposal
+
+### Who
+
+Initially this will rotate within the build v-team, but once we have better documentation (see later in doc) we may be able to expand it to the team as a whole.
+
+Since the focus is addressing urgent issues, we need **one person per time zone.**
+
+### Time commitment
+
+Shift duration will probably be either 1 or 2 weeks (need to discuss).
+
+Total time commitment would be allocated as **up to 25%**. _(DISCUSS: is this enough?)_
+
+Some weeks you'll get lucky and have almost nothing to do, but other weeks something will "blow up" in a way that requires much more time to address. If the time goes beyond ~25% _(or whatever number we decide above)_ it's worth calling out to management and/or pulling in other people as needed.
+
+### Responsibilities
+
+Starting out, the responsibility of build shield is limited to ensuring that PR/CI/release builds and local development stay unblocked (and documenting any failures that occur).
+
+#### Keeping builds unblocked
+
+This includes monitoring for:
+
+- Automated failure notifications in "Build failures (automated)" channel
+- Messages from team members (mainly in "Engineering Systems" channel) about PR or local build issues
+- Messages or github mentions from regular shield about urgent build-related regressions/breaks that showed up as github issues
+  - This is much less common and will primarily be up to the regular shield PM/dev to watch for these issues and loop in build shield
+
+Prioritization is roughly as follows. As with normal shield, it's fine to pull in others for consultation/help as needed, especially if it's a particularly complex or urgent issue, or if a more in-depth fix turns out to be necessary.
+
+| Issue type                                      | Priority                                                                     |
+| ----------------------------------------------- | ---------------------------------------------------------------------------- |
+| Release build failed                            | 🔥 Investigate ASAP (fix issue, or retry if intermittent)                    |
+| CI _consistently_ broken                        | 🔥 Fix ASAP                                                                  |
+| PR builds _consistently_ broken (CI okay)       | 🔥 Fix ASAP                                                                  |
+| Local builds _consistently_ broken (PR/CI okay) | 🔥 Fix ASAP                                                                  |
```

Comment:
For "consistently" and "intermittently" I was thinking of leaving to the person's best judgment, though I also see how specific guidelines or a process could be helpful. (Part of the goal of build shield is actually to build a "runbook" covering these type of guidelines, common failure types, and troubleshooting suggestions.)

As a starting point, basically if the CI build is failing more than 1-2x/day (which means PR builds are likely also failing => work is being blocked), that needs to be investigated as soon as reasonably possible. Priority increases if you see multiple failure notifications within an hour or two.

Or another way to think about it is that whenever there's an automated failure notification, it's build shield's responsibility to at least click the link and look at the error message ASAP (and add a note to the failure post about what the error was).
- If it's something that's likely to cause a failure every time and block PRs (like a snapshot issue), fix it yourself or find someone to fix it right away.
- If it's a known intermittent error (the `localDependencies` issue is a current example), no immediate action needed.
- Otherwise, up to your judgment. Look into it now if you have time, or wait and see if it happens again.

Note that PR builds (and local builds) are a little different because in those cases, there are no automated notifications and no way to meaningfully track metrics (PR builds are expected to fail a lot). So in that case you're relying on team members noticing and posting about issues.